Flaky test: test_menagerie_mujoco.TestMenagerie_BoosterT1.test_dynamics

## Description

`test_menagerie_mujoco.TestMenagerie_BoosterT1.test_dynamics` fails intermittently in CI when `qvel` diff exceeds the configured tolerance of `2e-5` due to float32 / GPU non-determinism.

The test itself already acknowledges this sensitivity at [`newton/tests/test_menagerie_mujoco.py:2157`](https://github.com/newton-physics/newton/blob/main/newton/tests/test_menagerie_mujoco.py#L2157):

```python
dynamics_tolerance = 2e-5  # float32 noise varies when running in batch
```

## Failure

```
AssertionError: Step 1, field 'qvel': max diff 4.824996e-05 > tol 2.000000e-05
```

## CI Run

https://github.com/newton-physics/newton/actions/runs/24733634589/job/72354874181

(branch: `ershi/fix-warp-1.11-api-removal`, commit `1cff17e9` — failure is unrelated to the branch's changes)

## Environment

- Instance: g7e.2xlarge (L40S GPU)
- HEAD: 1cff17e9dabc5e9db12a5f5e8421171ba9e7ebd4

## Suggested fix

Same pattern as PR #2493, which raised tolerances for sibling robots experiencing the same class of flake:

- `TestMenagerie_UnitreeG1`: `dynamics_tolerance = 1e-4  # GPU non-determinism: qvel diff up to 1.2e-5 across runs`
- `TestMenagerie_UnitreeGo2`: `dynamics_tolerance = 5e-4  # qvel drifts over steps; exact cause unclear`

BoosterT1 likely needs a comparable bump (e.g. `1e-4`) unless the underlying non-determinism can be pinned down.

Related: #1401 (Simulation correctness testing compared to native MjWarp).

--- a/newton/tests/test_menagerie_mujoco.py
+++ b/newton/tests/test_menagerie_mujoco.py
@@ -2154,7 +2154,7 @@ class TestMenagerie_BoosterT1(TestMenagerieMJCF):
 
     robot_folder = "booster_t1"
     num_steps = 20
-    dynamics_tolerance = 2e-5  # float32 noise varies when running in batch
+    dynamics_tolerance = 1e-4  # GPU atomic-reduction non-determinism: qvel diff up to 4.8e-5 observed on CI (#2526)
     fk_enabled = True
     backfill_model = True
 

